### ISSUE TITLE
Add "advert.rules" request parameter

### ISSUE THREAD

**Issue by jgrande** [salaboy/Solr-Advert#1]
This parameter tells the component which DRL file to use. The first time a file is referenced, the knowledge base is created and saved in a map, so it can be reused in the future.

**Comment by salaboy** [comment]
hmm we can create some sort of profiles.. or reuse the drools-spring configurations to name sessions and not use he drl file reference directly.

**Comment by jgrande** [comment]
Yeah, it's true that referencing the DRL file from the request is such a bad idea.

We can create profiles in the configuration of the component, and then reference them in the request. This is the way that the spellchecker works. For example, in solrconfig.xml you will have:

```
<searchComponent name="advert" class="com.plugtree.solradvert.AdvertComponent">
    <lst name="rules">
        <str name="name">foo</str>
        <str name="drl">foo.drl</str>
        <str name="drl">baz.drl</str>
    </lst>
    <lst name="rules">
        <str name="name">bar</str>
        <str name="drl">bar.drl</str>
        <str name="drl">baz.drl</str>
    </lst>
</search>
```

And then the request may be something like:

```
?q=shoes&advert=true&advert.rules=foo
```

I like this more than including Spring, because this is the way most Solr components work.

**Comment by salaboy** [comment]
We can do something like:
<lst name="rules">
      <drools:session name="myKsession"/>
 </lst>

In that way we will have the best way for both!

**Comment by jgrande** [comment]
@salaboy, please take a look at the droolsSpring branch. I implemented what you proposed in your previous comment. I'm sorry for putting my nose into your bussiness, but I wanted to know if I was understanding right what you were telling me!

So, in first place, when declaring the component in solrconfig.xml, you can point to a Spring context like this:

```
<searchComponent name="advert" class="com.plugtree.solradvert.AdvertComponent">
    <str name="kcontext">knowledge-service.xml</str>
</searchComponent>
```

In this context you must define a set of knowledge sessions like this:

```
<drools:kbase id="kbase1">
    <drools:resources>
        <drools:resource type="DRL" source="classpath:advert3.drl"/>
    </drools:resources>
</drools:kbase>

<drools:ksession id="ksession1" type="stateless" kbase="kbase1"/>
```

And then, you can point to any session in the request:

```
?q=shoes&advert=true&advert.rules=ksession1
```

What do you think about it?

**Comment by salaboy** [comment]
I'm not so sure about this:
<str name="kcontext">knowledge-service.xml</str>

I would like to have a more declarative approach.. where you put the name of the knowledge session (ksession id) and relate it with the search in the solr file. But I think that we can keep the solution that you propose for a while.. because it should be working good.

The next step is to change the DRL file to a pkg file, so when a search takes place the drools runtime don't need to compile anything. 
Good work man! don't hesitate to ask questions if you have.. I don't have too much time to look at this at the moment, but looks like you almost nail it!
Greetings.

**Comment by jgrande** [comment]
Closed in 5f8b85d6f42ef342a58b90f3df42d21eba62bc93, but may be optimized in the future!
